### ISSUE TITLE
[bitnami/redis] Redict Support for Redis Chart

### ISSUE THREAD

**Issue by gmemstr** [bitnami/charts#25164]
### Name and Version

bitnami/redis 19.1.0

### What is the problem this feature will solve?

Redis is [no longer free software](https://redis.io/blog/redis-adopts-dual-source-available-licensing/) and many are looking to adopt forks, including [Redict](https://redict.io/). Redict has taken the stance that Redis is more or less complete software, and will continue maintenence relatively unchanged. This means that going forward, there is a FOSS option for for Redis chart to use. Redict has changed the binary name to `redict-*`, which will require tweaks from the chart.

### What is the feature you are proposing to solve the problem?

Test and support either the upstream Redict image (https://codeberg.org/redict/-/packages/container/redict/7.3.0-bookworm) or build a custom image for the Helm chart.

**Comment by javsalgar** [comment]
Hi!

Thank you so much for the feature request. I will forward this to the team for evaluation.

**Comment by carrodher** [comment]
We're working on adding [Valkey](https://valkey.io/) to the catalog

**Comment by gmemstr** [comment]
Redict recently added compatibility images, which should be drop-in replacements for Redis images in this chart https://codeberg.org/redict/redict-containers/commit/91b0b245ce721a4cc5765417b646d97452a24d33 so I consider this complete to some degree :)

**Comment by AmitKatyal-Sophos** [comment]
Are we planning to add separate helm charts for valkey deployment as an alternative to redis helm chart ?

**Comment by javsalgar** [comment]
Hi! 

Yes, creating a helm chart for Valkey is in our roadmap

**Comment by AmitKatyal-Sophos** [comment]
@javsalgar Thanks for the update !
if we are already using redis helm charts with master-replica with sentinel configuration, the migration to valkey charts will be seamless ?

**Comment by carrodher** [comment]
That's a good question for the Valkey community, we highly recommend that you refer to the forums and user guides provided by the project responsible for the application itself since there you can find proper guidance about migration from Redis to Valkey.

**Comment by mickenordin** [comment]
It should be possible to just use the redict compatibility images with the redis-charts now:

https://redict.io/docs/install/containers/#compatibility

https://artifacthub.io/packages/helm/bitnami/redis#use-a-different-redis-version

**Comment by AmitKatyal-Sophos** [comment]
@mickenordin we have redict images for redis server only or the same container has redis-sentinel binary as well ?.
As per documentation, only scratch container has redis-sentinel binary. How about other variants (debian, alpine...etc)

**Comment by mickenordin** [comment]
> @mickenordin we have redict images for redis server only or the same container has redis-sentinel binary as well ?. As per documentation, only scratch container has redis-sentinel binary. How about other variants (debian, alpine...etc)

All of them have all the binaries. So if that is not clear, the docs should probably be updated. Can you point me to the doc that gave you the impression it was not there?

**Comment by AmitKatyal-Sophos** [comment]
@mickenordin Any specific reason that redict container images are not uploaded to the docker hub ?

**Comment by mickenordin** [comment]
> @mickenordin Any specific reason that redict container images are not uploaded to the docker hub ?

Yes, we discussed that and since people have to change their charts/configuration anyway, we prefer to run our own infrastructure, since it is something we know how to do and it will not add any additional inconvenience to people to just update the image to the full path http://registry.redict.io/redict.

**Comment by AmitKatyal-Sophos** [comment]
> > @mickenordin Any specific reason that redict container images are not uploaded to the docker hub ?
> 
> Yes, we discussed that and since people have to change their charts/configuration anyway, we prefer to run our own infrastructure, since it is something we know how to do and it will not add any additional inconvenience to people to just update the image to the full path registry.redict.io/redict.

we have a use-case of caching the image in our own hosted artifactory to avoid hitting public registry rate limiting, and unfortunately artifactory is only integrated with docker hub.

**Comment by mickenordin** [comment]
I think you can add any compliant docker registry to artefactory: https://medium.com/@andrenit/docker-and-artifactory-as-a-registry-mirror-1365891b39c5

**Comment by carrodher** [comment]
Bitnami just released the Valkey container images, see the images in DockerHub
- [bitnami/valkey](https://hub.docker.com/r/bitnami/valkey)
- [bitnami/valkey-sentinel](https://hub.docker.com/r/bitnami/valkey-sentinel)

as well as the source code in GitHub ([bitnami/valkey](https://github.com/bitnami/containers/tree/main/bitnami/valkey) and [bitnami/valkey-sentinel](https://github.com/bitnami/containers/tree/main/bitnami/valkey-sentinel))

Any feedback is more than welcome, you can report any issue or suggestion at https://github.com/bitnami/containers/issues/new/choose

Now it's time for the Helm chart which is being developed at https://github.com/bitnami/charts/pull/25643

**Comment by carrodher** [comment]
This issue has been inactive for a long time and is being closed as part of routine maintenance. Due to other higher-priority work, we weren’t able to investigate this further. If the issue is still relevant and you’re interested in contributing a fix, we’d be happy to review a pull request.

Thanks for your understanding.
